easybuild/easyblocks/n/nvhpc.py

```diff
@@ -0,0 +1,283 @@
+            'module_add_nccl':           [False, "Add NVHPC's NCCL library to module", CUSTOM],
+            'module_add_nvshmem':        [False, "Add NVHPC's NVSHMEM library to module", CUSTOM],
+            'module_add_profilers':      [False, "Add NVHPC's NVIDIA Profilers to module", CUSTOM]
+        }
+        return PackedBinary.extra_options(extra_vars)
+
+    def __init__(self, *args, **kwargs):
+        """Easyblock constructor, define custom class variables specific to NVHPC."""
+        super(EB_NVHPC, self).__init__(*args, **kwargs)
+
+        self.nvhpc_install_subdir = os.path.join('Linux_x86_64', self.version)
+
+    def install_step(self):
+        """Install by running install command."""
+
+        default_cuda_version = self.cfg['default_cuda_version']
+        if default_cuda_version is None:
+            module_cuda_version_full = get_software_version('CUDA')
+            if module_cuda_version_full is not None:
+                default_cuda_version = '.'.join(module_cuda_version_full.split('.')[:2])
+            else:
+                error_msg = "A default CUDA version is needed for installation of NVHPC. "
+                error_msg += "It can not be determined automatically and needs to be added manually. "
+                error_msg += "You can edit the easyconfig file, "
+                error_msg += "or use 'eb --try-amend=default_cuda_version=<version>'."
+                raise EasyBuildError(error_msg)
+
+        # Parse default_compute_capability from different sources (CLI has priority)
+        ec_default_compute_capability = self.cfg['cuda_compute_capabilities']
+        cfg_default_compute_capability = build_option('cuda_compute_capabilities')
+        if cfg_default_compute_capability is not None:
+            default_compute_capability = cfg_default_compute_capability
+        elif ec_default_compute_capability is not None:
+            default_compute_capability = ec_default_compute_capability
+        else:
+            error_msg = "A default Compute Capability is needed for installation of NVHPC."
+            error_msg += "Please provide it either in the easyconfig file like 'cuda_compute_capabilities=7.0',"
+            error_msg += "or use 'eb --cuda-compute-capabilities=7.0' from the command line."
+            raise EasyBuildError(error_msg)
+
+        # Extract first element of default_compute_capability list, if it is a list
+        if isinstance(default_compute_capability, list):
+            _before_default_compute_capability = default_compute_capability
+            default_compute_capability = _before_default_compute_capability[0]
+            print_warning(
+                "Replaced list of compute capabilities {before} with first element of list {after}"\
+                .format(
+                    before=_before_default_compute_capability,
+                    after=default_compute_capability
+                )
+            )
+        default_compute_capability.replace(".", "")
```

Comment:
Assignment is missing here, this can't work? `.replace` returns a new string value...

Also, I hate to be pedantic, but you're now just assuming you have a string value in `default_compute_capability`, which may not be the case (and if it's not, the crash will be pretty ugly).

So I'd add a small check:

```python
if isinstance(default_compute_capability, str):
    default_compute_capability = default_compute_capability.replace('.', '')
else:
    raise EasyBuildError("Unexpected non-string value encountered for compute capability: %s",
                         default_compute_capability)

Reply:
Implemented in 2c917098cb8c49ecff5888d1b17872fae6623d81